### PR TITLE
refactor: convert TextAreaSkeleton to TypeScript

### LINKED ISSUE #12560: Add TypeScript types to TextAreaSkeleton

### PR DESCRIPTION
**Pull request by sierrawetmore**
Closes #12560

This PR converts TextArea.Skeleton.js to typescript.

#### Changelog

**Changed**

- TextArea.Skeleton.js has been converted to TS

#### Testing / Reviewing

These changes can be tested and verified by running a fresh yarn, yarn build and yarn storybook. Verify that the behavior in the TextAreaSkeleton story remains unchanged

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

File: packages/react/src/components/TextArea/TextArea.Skeleton.tsx [hunk 1]
```diff
@@ -10,7 +10,21 @@ import React from 'react';
 import cx from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 
-const TextAreaSkeleton = ({ hideLabel, className, ...rest }) => {
+type ExcludedAttributes = '';
+export interface TextAreaSkeletonProps
+  extends Omit<React.InputHTMLAttributes<HTMLDivElement>, ExcludedAttributes> {
```
Comment (francinelucca) [inline packages/react/src/components/TextArea/TextArea.Skeleton.tsx]: is the omit necessary here? looks like no Attributes are being omitted
```suggestion
export interface TextAreaSkeletonProps
  extends React.InputHTMLAttributes<HTMLDivElement> {
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Mostly LGTM! one change suggestion

File: packages/react/src/components/TextArea/TextArea.Skeleton.tsx (hunk 1 above)
Comment (sierrawetmore) [reply packages/react/src/components/TextArea/TextArea.Skeleton.tsx]: You're correct, it isn't needed here. I've removed it.

Comment (francinelucca) [reply packages/react/src/components/TextArea/TextArea.Skeleton.tsx]: Hey @sierrawetmore looks like you removed it from `packages/react/src/components/TextArea/TextArea.tsx`, but I still see them on `packages/react/src/components/TextArea/TextArea.Skeleton.tsx`

**Review by francinelucca — CHANGES_REQUESTED** [review]

File: packages/react/src/components/TextArea/TextArea.Skeleton.tsx (hunk 1 above)
Comment (sierrawetmore) [reply packages/react/src/components/TextArea/TextArea.Skeleton.tsx]: Oops! Fixed it.

**Review by francinelucca — APPROVED** [review]

**Pull request merged**